cpp/unreal_plugins/SpEngine/Source/SpEngine/GameWorldService.h

```diff
@@ -36,6 +39,107 @@ struct GameWorldServicePropertyDesc
     MSGPACK_DEFINE_MAP(property_, value_ptr_);
 };
 
+/* Struct of optional parameters passed to SpawnActor function(s). */
```

Comment:
I'm surprised this struct doesn't cause a naming conflict with the existing Unreal struct? If I refer to `FActorSpawnParameters` in a block of code, how does the compiler know which struct I'm referring to? I think we should call this struct `FSpActorSpawnParameters` to avoid confusion. And for brevity, we should just provide the following top-level comment and remove the comments, which are not formatted correctly.

```
// This struct is intended to be identical to Unreal's FActorSpawnParameters struct, except we tag our struct
// as a USTRUCT so it is easier to pass to and from Python as a JSON string, see:
//     Engine/Source/...
```

Reply:
Updates:

- An inheritance-based solution doesn't work, even for trivial structs. The Unreal Build Tool generates errors because it wants the base struct to also be a `USTRUCT`.
- A composition-based solution will compile, but the problem is that the raw inner struct cannot be tagged as a `UPROPERTY` because its type is not a `USTRUCT`. Therefore, it will not be exposed to Unreal's property system, and by extension will not be exposed to our JSON serialization/deserialization functions. But if the inner struct cannot be set with JSON, then it is not useful in our setting.
- Based on these results, it looks like we'll need to do things the ugly verbose way. Note that everything in our `USTRUCT` will need to be declared as a `UPROPERTY`.

Reply:
Update:
Modified the code to do this the verbose way - defining a custom struct with USTRUCT() macro, where each property in this new struct corresponds to (or is a duplicate for) a property from the Unreal's struct.